### ISSUE TITLE
pygdf issue merging on 4 columns

### ISSUE THREAD

**Issue by brhodes10** [rapidsai/pygdf#281]
Seemed to hit a bug when trying to merge on 4 columns using pygdf

**Versions of things**
pygdf                     0.1.0a3+264.g6b7de97           <pip>
arrow-cpp                 0.10.0           py36h70250a7_0    conda-forge
pyarrow                   0.10.0           py36hfc679d8_0    conda-forge
numpy                     1.14.3           py36hcd700cb_1  
numpy-base                1.14.3           py36h9be14a7_1  
Python 3.6.6 :: Anaconda, Inc.
CUDA Version 9.2.148
Ubuntu 16.04.5 LTS (Xenial Xerus)
*Running from docker container

### **Auth (auth_pdf)**
**Data types**
 time             int64
auth_type     category
logon_type    category
auth_ori      category
suc_fail      category
src_user         int64
dest_user        int64
src_comp         int64
dest_comp        int64
dtype: object

  | time | auth_type | logon_type | auth_ori | suc_fail | src_user | dest_user | src_comp | dest_comp
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
0 | 4464508 | Kerberos | Network | LogOn | Success | 90403 | 90403 | 16419 | 16419
1 | 4464509 | Kerberos | Network | LogOn | Success | 90403 | 90403 | 4635 | 4635
2 | 4464511 | Kerberos | Network | LogOn | Success | 90403 | 90403 | 13054 | 13054
3 | 4464512 | Kerberos | Network | LogOn | Success | 90403 | 90403 | 8697 | 8697
4 | 4464513 | Kerberos | Network | LogOn | Success | 90403 | 90403 | 12727 | 12727

### **Redteam (redteam_pdf)**
Data types
time         int64
src_user     int64
src_comp     int64
dest_comp    int64
dtype: object

  | time | src_user | src_comp | dest_comp
-- | -- | -- | -- | --
0 | 150885 | 88893 | 5299 | 23
1 | 151036 | 92971 | 5299 | 12755
2 | 151648 | 92971 | 5299 | 15746
3 | 151993 | 88569 | 5299 | 1140
4 | 153792 | 89394 | 5299 | 12663

**Converting to gdf**
```
auth_gdf = DataFrame.from_pandas(auth_pdf)
redteam_gdf = DataFrame.from_pandas(redteam_pdf)
```

Merge

Pandas
```PASSED red_auth_pdf = pd.merge(auth_pdf, redteam_pdf, on=['src_user', 'src_comp', 'dest_comp', 'time'])```

Pygdf
```
FAILED red_auth_gdf = DataFrame.merge(auth_gdf, redteam_gdf, on=['src_user', 'src_comp', 'dest_comp', 'time'])
PASSED red_auth_gdf = DataFrame.merge(auth_gdf, redteam_gdf, on=['time'])
PASSED red_auth_gdf = DataFrame.merge(auth_gdf, redteam_gdf, on=['time', 'src_user'])
PASSED red_auth_gdf = DataFrame.merge(auth_gdf, redteam_gdf, on=['time', 'src_user', 'src_comp'])
FAILED red_auth_gdf = DataFrame.merge(auth_gdf, redteam_gdf, on=['time', 'src_user', 'src_comp', 'dest_comp'])
PASSED red_auth_gdf = DataFrame.merge(auth_gdf, redteam_gdf, on=['dest_comp'])
```

I don't see an error message but my jupyter kernel dies.

**Comment by kkraus14** [comment]
I believe this is due to conda build issues which are being addressed in https://github.com/rapidsai/libgdf/pull/167. If possible please try using the container from Dockerhub / NGC or install from source and see if you still see the same issues.

**Comment by brhodes10** [comment]
I've pulled the rapidsai container from dockerhub and ran jupyter-lab in debug mode. After running merge operation 
```
red_auth_gdf = DataFrame.merge(auth_gdf, redteam_gdf, on=['src_user', 'src_comp', 'dest_comp', 'time'], how='inner')
```
 The debug output showed the following:
```
python: /rapids/cudf-alpha/libgdf/src/join/hash/../../gdf_table.cuh:245: gdf_table<T, byte_t>::gdf_table(gdf_table<T, byte_t>::size_type, gdf_column**) [with T = long int; byte_t = unsigned char; gdf_table<T, byte_t>::size_type = long int; gdf_column = gdf_column_]: Assertion `num_cols > 0' failed.
```

**Comment by kkraus14** [comment]
@jrhemstad @kaatish See above error. Do we check if there's columns in addition to the join keys in the join kernel? Having all of the columns in the join key is valid.

**Comment by kaatish** [comment]
@kkraus14 Yeah, this looks like an error in libgdf. As of now, while computing the whole dataframe three gdf_tables are instantiated :
1. Left columns that are not joined on.
2. Common columns that are joined on.
3. Right columns that are not joined on.

I did not consider the case when either or both 1 and 3 are empty which might be leading to this error from the looks of it.

This should be an easy fix.

**Comment by jrhemstad** [comment]
@brhodes10 can you confirm that building libgdf from source from this PR fixes the issue you were seeing? https://github.com/rapidsai/libgdf/pull/172
